pkg/image/registry_options.go

```diff
@@ -0,0 +1,44 @@
+package image
+
+import (
+	"github.com/anchore/stereoscope/internal/log"
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+// RegistryOptions for the OCI registry provider.
+type RegistryOptions struct {
+	InsecureSkipTLSVerify bool
+	Credentials           []RegistryCredentials
+}
+
+// RegistryCredentials contains any information necessary to authenticate against an OCI-distribution-compliant
+// registry (either with basic auth or bearer token). Note: only valid for the OCI registry provider.
+type RegistryCredentials struct {
+	Authority string
+	Username  string
+	Password  string
+	Token     string
+}
+
+// Authenticator returns an object capable of authenticating against the given registry. If no credentials match the
+// given registry, or there is partial information configured, then nil is returned.
+func (r *RegistryOptions) Authenticator(registry string) authn.Authenticator {
+	for idx, auth := range r.Credentials {
+		if auth.Authority == registry {
```

Comment:
nit: _I_ tend to avoid nesting this much (for -> if -> if || else if) because it makes it trickier to read. One workaround is returning (continuing here I guess) when this condition _is not met_, which would allow the rest of the code to execute. That would outdent the rest, making it easier to read. Again, just a nit.

Reply:
ahh yeah, I consider that approach a "guard clause", I can switch to that 👍